### ISSUE TITLE
[exp-branch] Make generic

### ISSUE THREAD

**Issue by alexandru-dinu** [alexandru-dinu/pytorch-skel#4]
- [ ] encapsulate a training process as pointers to needed objects (i.e. dataloader, optimizer scheduler etc)
- [ ] use generic train function that makes use of the above
